src/simularium/VisGeometry/GeometryStore.ts

```diff
@@ -0,0 +1,358 @@
+import { forEach } from "lodash";
+import { OBJLoader } from "three/examples/jsm/loaders/OBJLoader";
+import jsLogger, { ILogger, ILogLevel } from "js-logger";
+import { BufferGeometry, Object3D, Mesh } from "three";
+
+import { checkAndSanitizePath } from "../../util";
+import PDBModel from "../PDBModel";
+import { InstancedMesh } from "../rendering/InstancedMesh";
+import VisAgent from "../VisAgent";
+import TaskQueue from "../worker/TaskQueue";
+import { AgentTypeVisData } from "../types";
+
+import {
+    AgentGeometry,
+    GeometryDisplayType,
+    GeometryStoreLoadResponse,
+    MeshGeometry,
+    MeshLoadRequest,
+} from "./types";
+
+export const DEFAULT_MESH_NAME = "SPHERE";
+
+type Registry = Map<string, AgentGeometry>;
+
+class GeometryStore {
+    private geoLoadAttempted: Map<string, boolean>;
+    private _cachedAssets: Map<string, string>;
+    private _registry: Registry;
+    public mlogger: ILogger;
+
+    constructor(loggerLevel?: ILogLevel) {
+        this.geoLoadAttempted = new Map<string, boolean>();
+        this._cachedAssets = new Map<string, string>();
+        this._registry = new Map<string, AgentGeometry>();
+        this.mlogger = jsLogger.get("geometry-store");
+        if (loggerLevel) {
+            this.mlogger.setLevel(loggerLevel);
+        }
+    }
+
+    public init(): void {
+        this._registry.clear();
+        this._registry.set(DEFAULT_MESH_NAME, {
+            displayType: GeometryDisplayType.SPHERE,
+            geometry: {
+                mesh: new Mesh(VisAgent.sphereGeometry),
+                cancelled: false,
+                instances: new InstancedMesh(
+                    VisAgent.sphereGeometry,
+                    DEFAULT_MESH_NAME,
+                    1
+                ),
+            },
+        });
+    }
+
+    public get registry(): Registry {
+        return this._registry;
+    }
+
+    public updateMeshes(): void {
```

Comment:
I think it's also reasonable for VisGeometry to do this, curious what people's opinions are